src/concern/mod.rs

```diff
@@ -103,6 +105,7 @@ pub struct WriteConcern {
 
 /// The type of the `w` field in a [`WriteConcern`](struct.WriteConcern.html).
 #[derive(Clone, Debug, PartialEq)]
+#[non_exhaustive]
```

Comment:
This may not be required here, since it would be a breaking change technically for the server to introduce a new special "w" value, given that any arbitrary string could refer to custom write concern created by a user.

Reply:
What about other types, like arrays or documents?

Reply:
Based on offline discussion, we'll leave this as-is